### PR TITLE
Add status_filters to pr resource

### PR DESCRIPTION
**Pull request by defjosiah**
We're over-running concourseci PRs, since we're taking every single PR/commit (even when we expect it to settle down after tests run, etc.). 

This extends the PR resource as follows:
```yaml
resource_types:
  - name: pull-request
    type: docker-image
    source:
      repository: opendoor/telia-oss-github-pr-resource
resources:
  - name: pr-branch
    type: pull-request
    source:
      repository: opendoor-labs/code
      access_token: ((github-token))
      base_branch: master
      status_filters:
        - context: 'ci/circleci: js_lint'
          state: 'success'
```

You can now add an optional parameter (status_filters) and specify an array of 
"context" (githubs name for a status check result) and "state" params. This will 
now only select PRs that have that desired state for a particular check.

In the example above, this would only select PRs that have passed the js_lint step in 
circleci.

Testing:
I added tests, I also ran the check locally to verify that it worked (and it did).

`go run ./cmd/check/main.go` and then pasting in the "source:" field as json as stdin

### CODE REVIEW

File: check.go [hunk 1]
```diff
@@ -57,6 +57,22 @@ Loop:
 			}
 		}
 
+		if len(request.Source.StatusFilters) > 0 {
+			for _, statusFilter := range request.Source.StatusFilters {
+				isValid := false
+				for _, prStatus := range p.Tip.Status.Contexts {
+					if prStatus.Context == statusFilter.Context && strings.EqualFold(prStatus.State, statusFilter.State) {
+						// requires both that status and that that status matches the desired state
```
Comment (nathanshelly) [inline check.go:65]: is this saying requires that there is a status?

**Review by lxe — APPROVED** [review]

File: check.go [hunk 2]
```diff
@@ -57,6 +57,22 @@ Loop:
...
+						isValid = true
+						break
+					}
+				}
+				if !isValid {
+					continue Loop
+				}
```
Comment (nathanshelly) [inline check.go:72]: does using a continue like this mean that Go will implicitly break if the continue doesn't happen?

**Review by nathanshelly — APPROVED** [review]
great tests!

File: check.go (hunk 2 above)
Comment (defjosiah) [reply check.go:72]: Their structure is a bit odd (but makes sense), there’s a loop with a label. So this line is saying to continue in the outer loop (skipping that PR) 
If it’s all valid, then it’ll just finish this inner loop and go to the next clause

File: check.go (hunk 1 above)
Comment (defjosiah) [reply check.go:65]: Yeah, I debated this, but I think if you add one of these status filters, then it should also reject anything without that status

Comment (nathanshelly) [reply check.go:65]: gotcha, thoughts on this rephrasing?

```suggestion
						// requires the given status exists and that it matches the desired state
```

File: check.go (hunk 2 above)
Comment (nathanshelly) [reply check.go:72]: ah got it I was wondering where that `Loop` came from and just assumed it was some primitive referring to the current loop. thanks for the explanation!

**Review by nathanshelly — APPROVED** [review]

**Pull request merged**